### ISSUE TITLE
bug(engine): experimental-queries.json: no such file or directory

### ISSUE THREAD

**Issue by LvffY** [Checkmarx/kics#6753]
### Expected Behavior

We'd like to be able to run a Kics analysis with kics **1.7.9** as we do with previous versions (like **1.7.8**).

### Actual Behavior

     Note: The exact same configurations works if we're using the checkmarcks/kics:v1.7.8-debian

We followed the [Azure devops integration](https://docs.kics.io/latest/integrations_azurepipelines/) and have the following logs :

```text
2023-10-03T07:24:35.6802924Z ##[section]Starting: Run kics analysis
2023-10-03T07:24:35.6807617Z ==============================================================================
2023-10-03T07:24:35.6807739Z Task         : Bash
2023-10-03T07:24:35.6807790Z Description  : Run a Bash script on macOS, Linux, or Windows
2023-10-03T07:24:35.6807898Z Version      : 3.227.0
2023-10-03T07:24:35.6807974Z Author       : Microsoft Corporation
2023-10-03T07:24:35.6808043Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/bash
2023-10-03T07:24:35.6808152Z ==============================================================================
2023-10-03T07:24:36.0562611Z Generating script.
2023-10-03T07:24:36.0607982Z Script contents:
2023-10-03T07:24:36.0608533Z /app/bin/kics scan --ci -p ${PWD} -o ${PWD}/kics --report-formats sarif,sonarqube --ignore-on-exit results
2023-10-03T07:24:36.0615614Z ========================== Starting Command Output ===========================
2023-10-03T07:24:36.0621297Z [command]/bin/bash /__w/_temp/ab37e812-21c3-4e42-8c9e-5b66f93753be.sh
2023-10-03T07:24:36.1230575Z 7:24AM INF Scanning with Keeping Infrastructure as Code Secure v1.7.9
2023-10-03T07:24:36.1231073Z 7:24AM INF Operating system: linux
2023-10-03T07:24:36.1231378Z 7:24AM INF Total memory: 6.8G
2023-10-03T07:24:36.1231697Z 7:24AM INF CPU: 2.0
2023-10-03T07:24:36.3194210Z 7:24AM INF Total files in the project: 2021
2023-10-03T07:24:36.3204245Z 7:24AM INF .gitignore file was found in '/__w/1/s' and it will be used to automatically exclude paths
2023-10-03T07:24:37.1161959Z 7:24AM INF Loading queries of type: ansible, googledeploymentmanager
2023-10-03T07:24:37.1171374Z Error: stat /__w/1/s/assets/utils/experimental-queries.json: no such file or directory
2023-10-03T07:24:37.1172252Z Usage:
2023-10-03T07:24:37.1173759Z   kics scan [flags]
2023-10-03T07:24:37.1174123Z 
2023-10-03T07:24:37.1174485Z Flags:
2023-10-03T07:24:37.1175337Z   -m, --bom                            include bill of materials (BoM) in results output
2023-10-03T07:24:37.1175699Z       --cloud-provider strings         list of cloud providers to scan (alicloud, aws, azure, gcp)
2023-10-03T07:24:37.1175980Z       --config string                  path to configuration file
2023-10-03T07:24:37.1176288Z       --disable-full-descriptions      disable request for full descriptions and use default vulnerability descriptions
2023-10-03T07:24:37.1176572Z       --disable-secrets                disable secrets scanning
2023-10-03T07:24:37.1176840Z       --exclude-categories strings     exclude categories by providing its name
2023-10-03T07:24:37.1177021Z                                        cannot be provided with query inclusion flags
2023-10-03T07:24:37.1177205Z                                        can be provided multiple times or as a comma separated string
2023-10-03T07:24:37.1177450Z                                        example: 'Access control,Best practices'
2023-10-03T07:24:37.1177745Z       --exclude-gitignore              disables the exclusion of paths specified within .gitignore file
2023-10-03T07:24:37.1177998Z   -e, --exclude-paths strings          exclude paths from scan
2023-10-03T07:24:37.1178198Z                                        supports glob and can be provided multiple times or as a quoted comma separated string
2023-10-03T07:24:37.1178459Z                                        example: './shouldNotScan/*,somefile.txt'
2023-10-03T07:24:37.1178723Z       --exclude-queries strings        exclude queries by providing the query ID
2023-10-03T07:24:37.1178889Z                                        cannot be provided with query inclusion flags
2023-10-03T07:24:37.1179459Z                                        can be provided multiple times or as a comma separated string
2023-10-03T07:24:37.1179816Z                                        example: 'e69890e6-fce5-461d-98ad-cb98318dfc96,4728cd65-a20c-49da-8b31-9c08b423e4db'
2023-10-03T07:24:37.1180218Z   -x, --exclude-results strings        exclude results by providing the similarity ID of a result
2023-10-03T07:24:37.1180670Z                                        can be provided multiple times or as a comma separated string
2023-10-03T07:24:37.1181007Z                                        example: 'fec62a97d569662093dbb9739360942f...,31263s5696620s93dbb973d9360942fc2a...'
2023-10-03T07:24:37.1181323Z       --exclude-severities strings     exclude results by providing the severity of a result
2023-10-03T07:24:37.1181509Z                                        can be provided multiple times or as a comma separated string
2023-10-03T07:24:37.1181748Z                                        example: 'info,low'
2023-10-03T07:24:37.1181997Z       --exclude-type strings           case insensitive list of platform types not to scan
2023-10-03T07:24:37.1182293Z                                        (Ansible, AzureResourceManager, Buildah, CICD, CloudFormation, Crossplane, DockerCompose, Dockerfile, GRPC, GoogleDeploymentManager, Knative, Kubernetes, OpenAPI, Pulumi, ServerlessFW, Terraform)
2023-10-03T07:24:37.1182583Z                                        cannot be provided with type inclusion flags
2023-10-03T07:24:37.1182944Z       --experimental-queries strings   include experimental queries (queries not yet thoroughly reviewed) by providing the path of the queries folder
2023-10-03T07:24:37.1183349Z                                        can be provided multiple times or as a comma-separated string (platform/cloudProvider or platform)
2023-10-03T07:24:37.1183606Z                                        example: 'terraform/databricks'
2023-10-03T07:24:37.1183893Z                                        possible values found in: '/assets/utils/experimental-queries.json'
2023-10-03T07:24:37.1184187Z       --fail-on strings                which kind of results should return an exit code different from 0
2023-10-03T07:24:37.1184375Z                                        accepts: high, medium, low and info
2023-10-03T07:24:37.1184526Z                                        example: "high,low" (default [high,medium,low,info])
2023-10-03T07:24:37.1184753Z   -h, --help                           help for scan
2023-10-03T07:24:37.1185013Z       --ignore-on-exit string          defines which kind of non-zero exits code should be ignored
2023-10-03T07:24:37.1185202Z                                        accepts: all, results, errors, none
2023-10-03T07:24:37.1185527Z                                        example: if 'results' is set, only engine errors will make KICS exit code different from 0 (default "none")
2023-10-03T07:24:37.1185837Z   -i, --include-queries strings        include queries by providing the query ID
2023-10-03T07:24:37.1186025Z                                        cannot be provided with query exclusion flags
2023-10-03T07:24:37.1186189Z                                        can be provided multiple times or as a comma separated string
2023-10-03T07:24:37.1186534Z                                        example: 'e69890e6-fce5-461d-98ad-cb98318dfc96,4728cd65-a20c-49da-8b31-9c08b423e4db'
2023-10-03T07:24:37.1187111Z       --input-data string              path to query input data files
2023-10-03T07:24:37.1187450Z   -b, --libraries-path string          path to directory with libraries (default "./assets/libraries")
2023-10-03T07:24:37.1187968Z       --minimal-ui                     simplified version of CLI output
2023-10-03T07:24:37.1188284Z       --no-progress                    hides the progress bar
2023-10-03T07:24:37.1188539Z       --output-name string             name used on report creations (default "results")
2023-10-03T07:24:37.1188806Z   -o, --output-path string             directory path to store reports
2023-10-03T07:24:37.1189197Z   -p, --path strings                   paths or directories to scan
2023-10-03T07:24:37.1189365Z                                        example: "./somepath,somefile.txt"
2023-10-03T07:24:37.1189648Z       --payload-lines                  adds line information inside the payload when printing the payload file
2023-10-03T07:24:37.1189952Z   -d, --payload-path string            path to store internal representation JSON file
2023-10-03T07:24:37.1190376Z       --preview-lines int              number of lines to be display in CLI results (min: 1, max: 30) (default 3)
2023-10-03T07:24:37.1190702Z   -q, --queries-path strings           paths to directory with queries (default [./assets/queries])
2023-10-03T07:24:37.1191155Z       --report-formats strings         formats in which the results will be exported (all, asff, codeclimate, csv, cyclonedx, glsast, html, json, junit, pdf, sarif, sonarqube) (default [json])
2023-10-03T07:24:37.1191506Z   -r, --secrets-regexes-path string    path to secrets regex rules configuration file
2023-10-03T07:24:37.1191800Z       --terraform-vars-path string     path where terraform variables are present
2023-10-03T07:24:37.1192110Z       --timeout int                    number of seconds the query has to execute before being canceled (default 60)
2023-10-03T07:24:37.1192405Z   -t, --type strings                   case insensitive list of platform types to scan
2023-10-03T07:24:37.1192681Z                                        (Ansible, AzureResourceManager, Buildah, CICD, CloudFormation, Crossplane, DockerCompose, Dockerfile, GRPC, GoogleDeploymentManager, Knative, Kubernetes, OpenAPI, Pulumi, ServerlessFW, Terraform)
2023-10-03T07:24:37.1192962Z                                        cannot be provided with type exclusion flags
2023-10-03T07:24:37.1193031Z 
2023-10-03T07:24:37.1193145Z Global Flags:
2023-10-03T07:24:37.1193398Z       --ci                  display only log messages to CLI output (mutually exclusive with silent)
2023-10-03T07:24:37.1193697Z   -f, --log-format string   determines log format (pretty,json) (default "pretty")
2023-10-03T07:24:37.1193999Z       --log-level string    determines log level (TRACE,DEBUG,INFO,WARN,ERROR,FATAL) (default "INFO")
2023-10-03T07:24:37.1194272Z       --log-path string     path to generate log file (info.log)
2023-10-03T07:24:37.1194485Z       --no-color            disable CLI color output
2023-10-03T07:24:37.1194834Z       --profiling string    enables performance profiler that prints resource consumption metrics in the logs during the execution (CPU, MEM)
2023-10-03T07:24:37.1195156Z   -s, --silent              silence stdout messages (mutually exclusive with verbose and ci)
2023-10-03T07:24:37.1195449Z   -v, --verbose             write logs to stdout too (mutually exclusive with silent)
2023-10-03T07:24:37.1195534Z 
2023-10-03T07:24:37.1195898Z 7:24AM ERR Failed to run application error="stat /__w/1/s/assets/utils/experimental-queries.json: no such file or directory"
2023-10-03T07:24:37.1300168Z ##[error]Bash exited with code '126'.
2023-10-03T07:24:37.1382969Z ##[section]Finishing: Run kics analysis
```

### Steps to Reproduce the Problem

1.  `/app/bin/kics scan --ci -p ${PWD} -o ${PWD}/kics --report-formats sarif,sonarqube --ignore-on-exit results` in the **checkmarcks/kics:debian** container in an Azure DevOps pipeline

### Specifications
(N/A if not applicable)

-   Version: 1.7.9
-   Platform: Linux
-   Subsystem: Ubuntu

We run on the **checkmarcks/kics:debian** container.

**Comment by army-watchfinder** [comment]
We have the same issue here on Ubuntu latest and Azure Pipelines 4

**Comment by gabriel-cx** [comment]
Hi @LvffY @army-watchfinder 

Thanks for the info.
We are working on it. We will come up with a fix asap.

**Comment by gabriel-cx** [comment]
@LvffY @army-watchfinder, a new [docker image was published](https://hub.docker.com/layers/checkmarx/kics/v1.7.9-debian/images/sha256-fe93d547d4fe1236b81128ef762ff36b0069b6033472df885f902cf0bf720f67?context=explore) today with the fix. Kindly re-try from your side. It should work as expected. :) 

If it works also on your side, kindly close this issue.
Otherwise feel free to leave us a message requesting support.

**Comment by LvffY** [comment]
Hi @gabriel-cx this seems to work with **checkmarx/kics:v1.7.9-debian**.

I would just like to have a **latest** image with the fix before closing the issue. For Azure DevOps **checkmarx/kics:debian** would be enough, but I think that a rebuild of all images could be a help for many people :)

**Comment by gabriel-cx** [comment]
Hi @LvffY ,

Sure! We plan to release a new latest image this week (asap) with the fix.

**Comment by nvuillam** [comment]
Same issue with MegaLinter, I hope the fix will be released soon in docker image `checkmarx/kics:alpine` ;)

**Comment by gabriel-cx** [comment]
Hi @nvuillam ,

Yes, we are working on it. We will release it asap.

**Comment by gabriel-cx** [comment]
Hi @LvffY @army-watchfinder  @nvuillam ,

A [new KICS version](https://github.com/Checkmarx/kics/releases/tag/v1.7.10) was released just now!
The correct behavior should now be available in all the environments.

If everything is okey, kindly close this issue from your side.
If not, leave us a message and we will tackle the issue you are having.

**Comment by LvffY** [comment]
@gabriel-cx

**Comment by nvuillam** [comment]
@LvffY sorry but the bug is still happening on my side :/

```sh
❌ Linted [REPOSITORY] files with [kics]: Found 1 error(s) - (0.51s)
- Using [kics v1.7.10] https://megalinter.io/test-nuillam-merge/descriptors/repository_kics
- MegaLinter key: [REPOSITORY_KICS]
- Rules config: identified by [kics]
--Error detail:

                   .0MO.                                    
                   OMMMx                                    
                   ;NMX;                                    
                    ...           ...              ....     
WMMMd     cWMMM0.  KMMMO      ;xKWMMMMNOc.     ,xXMMMMMWXkc.
WMMMd   .0MMMN:    KMMMO    :XMMMMMMMMMMMWl   xMMMMMWMMMMMMl
WMMMd  lWMMMO.     KMMMO   xMMMMKc...'lXMk   ,MMMMx   .;dXx 
WMMMd.0MMMX;       KMMMO  cMMMMd        '    'MMMMNl'       
WMMMNWMMMMl        KMMMO  0MMMN               oMMMMMMMXkl.  
WMMMMMMMMMMo       KMMMO  0MMMX                .ckKWMMMMMM0.
WMMMMWokMMMMk      KMMMO  oMMMMc              .     .:OMMMM0
WMMMK.  dMMMM0.    KMMMO   KMMMMx'    ,kNc   :WOc.    .NMMMX
WMMMd    cWMMMX.   KMMMO    kMMMMMWXNMMMMMd .WMMMMWKO0NMMMMl
WMMMd     ,NMMMN,  KMMMO     'xNMMMMMMMNx,   .l0WMMMMMMMWk, 
xkkk:      ,kkkkx  okkkl        ;xKXKx;          ;dOKKkc    


Scanning with Keeping Infrastructure as Code Secure v1.7.10


Error: stat /tmp/lint/.automation/test/repository_kics/assets/utils/experimental-queries.json: no such file or directory
Usage:
  kics scan [flags]
```

Full log: https://github.com/oxsecurity/megalinter/actions/runs/6474361071/job/17579143896?pr=2997

**Comment by nvuillam** [comment]
You seem to have some code that looks in the relative path of your docker image, not the absolute one

**Comment by pereiramarco011** [comment]
Hello @nvuillam,

Thanks for bringing this to our attention, while looking into your Dockerfile configurations it seems that you are saving the assets to a folder not in the same directory as the KICS engine. Currently KICS will look on its directory for the experimental queries.json file. What we would ask is if it is possible for you to structure your Dockerfile configurations so that the structure of KICS remains the same as the team intended. 

My suggestion would be:
```Dockerfile
FROM checkmarx/kics:alpine as kics
COPY --link --from=kics /app/bin/kics /usr/bin/
ENV KICS_QUERIES_PATH=/usr/bin/kics/assets/queries KICS_LIBRARIES_PATH=/usr/bin/kics/assets/libraries
COPY --from=kics /app/bin/assets /usr/bin/
```

We will be discussing the possibility of adding an environment variable for the experimental queries path over the next releases.

Best Regards,

KICS Team

**Comment by nvuillam** [comment]
I will try that tonight, many thanks for the support :)

**Comment by army-watchfinder** [comment]
> @gabriel-cx We tested it today, it seems to work fine 😃
> 
> Thanks for the quick fix !

Just as a follow-up, it works for us as well. Good work, thanks!

**Comment by nvuillam** [comment]
@pereiramarco011 I have the following error with your dockerfile instructions  :/

```
Scanning with Keeping Infrastructure as Code Secure v1.7.10


Error: stat /usr/bin/kics/assets/queries: not a directory
```

**Comment by pereiramarco011** [comment]
Hello @nvuillam,

I apologize, I placed the field in a directory and pointed to the wrong one, here is another suggestion:

```Dockerfile
FROM checkmarx/kics:alpine as kics
COPY --link --from=kics /app/bin/kics /usr/bin/
ENV KICS_QUERIES_PATH=/usr/bin/assets/queries KICS_LIBRARIES_PATH=/usr/bin/assets/libraries
COPY --from=kics /app/bin/assets /usr/bin/
```

If this suggestion still doesn't work, my other suggestion would be:

```Dockerfile
FROM checkmarx/kics:alpine as kics
COPY --link --from=kics /app/bin/kics /usr/bin/
ENV KICS_QUERIES_PATH=/usr/bin/kics/assets/queries KICS_LIBRARIES_PATH=/usr/bin/kics/assets/libraries
COPY --from=kics /app/bin/assets /usr/bin/kics/
```

Please try it and give me some feedback ;)

**Comment by nvuillam** [comment]
First suggestion result: 
![image](https://github.com/Checkmarx/kics/assets/17500430/c77ef6b8-4145-4709-8f52-090b9cdc31c5)

Now trying the second one !

**Comment by nvuillam** [comment]
Second attempt failed :/ https://github.com/oxsecurity/megalinter/actions/runs/6508613773/job/17678360163?pr=3001

```sh
10 [kics 1/1] FROM docker.io/checkmarx/kics:alpine@sha256:f4500fece706e5dba3e273ba7c14d470d73c630f62f30f39c[297](https://github.com/oxsecurity/megalinter/actions/runs/6508613773/job/17678360163?pr=3001#step:7:302)f30ee669ecde
#10 extracting sha256:0469bc75114a3f0cfd4559f7aec03ba69556ec500351bb6dbecf44e79040265c done
#10 extracting sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1 done
#10 DONE 14.9s

#15 [stage-1  4/18] COPY --link --from=kics /app/bin/kics /usr/bin/
#15 merging 0.0s done
#15 DONE 0.2s

#16 [stage-1  5/18] COPY --from=kics /app/bin/assets /usr/bin/kics/
#16 ERROR: mkdir /var/lib/buildkit/runc-overlayfs/cachemounts/buildkit1270371276/usr/bin/kics: not a directory
------
 > [stage-1  5/18] COPY --from=kics /app/bin/assets /usr/bin/kics/:
------
Dockerfile:118
--------------------
 116 |     #COPY__START
 117 |     COPY --link --from=kics /app/bin/kics /usr/bin/
 118 | >>> COPY --from=kics /app/bin/assets /usr/bin/kics/
 119 |     #COPY__END
 120 |     
--------------------
ERROR: failed to solve: mkdir /var/lib/buildkit/runc-overlayfs/cachemounts/buildkit1270371276/usr/bin/kics: not a directory
```

**Comment by pereiramarco011** [comment]
Can you try the following?

```Dockerfile
FROM checkmarx/kics:alpine as kics
COPY --link --from=kics /app/bin/kics /usr/bin/kics
ENV KICS_QUERIES_PATH=/usr/bin/assets/queries KICS_LIBRARIES_PATH=/usr/bin/assets/libraries
COPY --from=kics /app/bin/assets /usr/bin/assets
```

If this doesn't work, on monday I will try to provide a better alternative. Over the next releases a better fix will be provided that won't require modifying your own configuration.

Thanks for understanding and being cooperative,

KICS Team

**Comment by nvuillam** [comment]
@pereiramarco011 it worked, many thanks for your support, now I can reactivate KICS in MegaLinter ;)

**Comment by pereiramarco011** [comment]
@nvuillam 

Awesome, good to know, over the next releases we will provide a better fix, after that is released you can go back to you previous dockerfile configuration if you want 😄 

Best Regards,

KICS Team

**Comment by vjeantet** [comment]
Hello, I still have the issue using the brew version.

I'm using the last version available 1.7.10.

How can I help ?

**Comment by gabriel-cx** [comment]
Hi @baptfh @vjeantet ,

We released a new KICS version ([v1.7.11](https://github.com/Checkmarx/kics/releases/tag/v1.7.11)). Kindly use the latest release and this problem you mentioned should now be gone!
Kindly let us know if now everything works as expected in your side.
